CONTRIBUTING-SAIL-PROJECT.md

```diff
@@ -0,0 +1,51 @@
+# Contribution guidelines to the sail-operator project
```

Comment:
I just verified that we can call this file `CONTRIBUTING.md` and it will not be overwritten by `make update-common` as long as it doesn't link to `istio/community/blob/master/CONTRIBUTING.md`